ocaml/testsuite/tests/typing-unique/overwriting_proj_push_down_bug.ml

```diff
@@ -376,15 +392,69 @@ Line 3, characters 41-42:
 
 |}]
 
-(* CR uniqueness: Update this test once overwriting is fully implemented.
-let swap_inner (t : tree) =
-  match t with
-  | Node { l = Node { r = lr } as l; r = Node { l = rl } as r } as t ->
-      overwrite_ t with
-        Node { l = overwrite_ l with Node { r = rl; };
-               r = overwrite_ r with Node { l = lr; }}
-  | _ -> t
+(********************************************)
+(* Global allocations in overwritten fields *)
```

Comment:
Yes, we can not run this test yet, because we can't see the generated lambda without promoting overwriting into lambda. This test should ensure that allocations made in a overwrite are actually allocated on the heap. This is the test you asked for on slack, since it would not be good enough to just mark these allocations as global.

Reply:
Ah -- I think I had been expecting the input to these functions to be marked as local, because then it's a little more challenging to recognize that the allocations must still be on the heap.